### PR TITLE
fix: empty ADSCESC1 composition path edge case

### PR DESCRIPTION
**Pull request by mistahj67**
## Description

Fixed the path traverse logic to find the first enterprise CA instead of the last enterprise CA in the case that there are multiple in 1 pathway.

## Motivation and Context

In some cases, viewing the composition for an `ADSCESC1` edge would result in am empty path and query "failed" message. 

## How Has This Been Tested?

A / B testing locally with reproducible data
Added integration test

## Types of changes

-   [ ] Chore (a change that does not modify the application functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Documentation updates are needed, and have been made accordingly.
-   [x] I have added and/or updated tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] My changes include a database migration.

### CODE REVIEW

File: packages/go/analysis/ad/ad.go [hunk 1]
```diff
@@ -525,57 +525,33 @@ func getCardinalityCount(id uint32, expansions cardinality.Provider[uint32], car
...
-				return err
-			} else {
-				pathSet = results
-			}
+	var (
+		err     error
+		pathSet = graph.NewPathSet()
+	)
+
+	if err = db.ReadTransaction(ctx, func(tx graph.Transaction) error {
```
Comment (mistahj67) [inline packages/go/analysis/ad/ad.go:533]: No changes here, just cleanup for readability

File: cmd/api/src/test/integration/harnesses.go [hunk 2]
```diff
@@ -1224,6 +1225,7 @@ func (s *ADCSESC1Harness) Setup(graphTestContext *GraphTestContext) {
 	s.AuthStore2 = graphTestContext.NewActiveDirectoryNTAuthStore("authstore2", sid)
 	s.EnterpriseCA21 = graphTestContext.NewActiveDirectoryEnterpriseCA("eca2-1", sid)
 	s.EnterpriseCA22 = graphTestContext.NewActiveDirectoryEnterpriseCA("eca2-2", sid)
+	s.EnterpriseCA23 = graphTestContext.NewActiveDirectoryEnterpriseCA("eca2-3", sid)
```
Comment (mistahj67) [inline cmd/api/src/test/integration/harnesses.go:1228]: This addition to the harness replicates the edge case to prevent this going forward

**Comment by rvazarkar** [comment]
This all looks good to me, but I helped work on some of the logic a bit, so would love some other eyes as well

File: packages/go/analysis/ad/ad.go (hunk 1 above)
Comment (urangel) [reply packages/go/analysis/ad/ad.go:533]: nice 👍

File: cmd/api/src/analysis/ad/adcs_integration_test.go [hunk 3]
```diff
@@ -100,8 +99,57 @@ func TestADCSESC1(t *testing.T) {
...
+				)
+			}).First(); err != nil {
+				t.Fatalf("error fetching esc1 edges in integration test; %v", err)
+			} else {
+				comp, err := ad2.GetADCSESC1EdgeComposition(context.Background(), db, edge)
+				assert.Nil(t, err)
+
+				esc13nodes := comp.AllNodes()
+				assert.Len(t, esc13nodes, 6)
+				require.True(t, esc13nodes.Contains(harness.ADCSESC1Harness.Group32))
```
Comment (urangel) [inline cmd/api/src/analysis/ad/adcs_integration_test.go:142]: nit: I thought this was related to escalation 13 from the var name. Maybe we can name these `domain3Nodes` or something instead?

**Review by urangel — APPROVED** [review]

File: cmd/api/src/analysis/ad/adcs_integration_test.go (hunk 3 above)
Comment (mistahj67) [reply cmd/api/src/analysis/ad/adcs_integration_test.go:142]: Ha, fun coincidence, will update!
